ignite/metrics/gan/fid.py

```diff
@@ -0,0 +1,207 @@
+            :class:`~ignite.engine.engine.Engine`'s ``process_function``'s output into the
+            form expected by the metric. This can be useful if, for example, you have a multi-output model and
+            you want to compute the metric with respect to one of the outputs.
+            By default, metrics require the output as ``(y_pred, y)`` or ``{'y_pred': y_pred, 'y': y}``.
+        device: specifies which device updates are accumulated on. Setting the
+            metric's device to be the same as your ``update`` arguments ensures the ``update`` method is
+            non-blocking. By default, CPU.
+
+    Example:
+
+        .. code-block:: python
+
+            from ignite.metric.gan import FID
+            import torch
+            y_pred, y = torch.rand(10,2048), torch.rand(10,2048)
+            m = FID()
+            m.update((y_pred,y))
+            print(m.compute())
+
+    .. versionadded:: 0.5.0
+    """
+
+    def __init__(
+        self,
+        num_features: Optional[int] = None,
+        feature_extractor: Optional[Callable] = None,
+        output_transform: Callable = lambda x: x,
+        device: Union[str, torch.device] = torch.device("cpu"),
+    ) -> None:
+
+        try:
+            import scipy  # noqa: F401
+        except ImportError:
+            raise RuntimeError("This module requires scipy to be installed.")
+
+        # default is inception
+        if num_features is None and feature_extractor is None:
+            num_features = 2048
+            feature_extractor = InceptionExtractor()
+        elif num_features is None:
+            raise ValueError("num of features should be defined")
+        elif feature_extractor is None:
+            self._feature_extractor = lambda x: x
+            feature_extractor = self._feature_extractor
+
+        if num_features <= 0:
+            raise ValueError(f"num of features must be greater to zero (got: {num_features})")
+        self._num_features = num_features
+        self._feature_extractor = feature_extractor
+        self._eps = 1e-6
+        super(FID, self).__init__(output_transform=output_transform, device=device)
+
+    @staticmethod
+    def _online_update(features: torch.Tensor, total: torch.Tensor, sigma: torch.Tensor) -> None:
+        total += features
+        sigma += torch.outer(features, features)
+
+    def get_covariance(self, sigma: torch.Tensor, total: torch.Tensor) -> torch.Tensor:
+        r"""
+        Calculates covariance from mean and sum of products of variables
+        """
+        sub_matrix = torch.outer(total, total)
+        sub_matrix = sub_matrix / self._num_examples
+        return (sigma - sub_matrix) / (self._num_examples - 1)
+
+    @staticmethod
+    def _check_feature_input(train: torch.Tensor, test: torch.Tensor) -> None:
+        for feature in [train, test]:
+            if feature.dim() != 2:
+                raise ValueError(f"Features must be a tensor of dim 2 (got: {feature.dim()})")
```

Comment:
@gucifer we usually does "..., got {}". Please update all messages for this metric in this PR and for nlp/rouge in another PR. We have to keep consistency for the codebase.